src/openloco/graphics/gfx.cpp

```diff
@@ -705,37 +730,38 @@ namespace openloco::gfx
     }
 
     // 0x004474BA
-    static void draw_rect_impl(gfx::drawpixelinfo_t* dpi, int16_t left, int16_t top, int16_t right, int16_t bottom, uint32_t color)
+    static void draw_rect_impl(gfx::drawpixelinfo_t* dpi, int16_t left, int16_t top, int16_t right, int16_t bottom, uint32_t colour, uint8_t flags)
     {
         registers regs;
         regs.ax = left;
         regs.bx = right;
         regs.cx = top;
         regs.dx = bottom;
-        regs.ebp = color;
+        regs.ebp = colour;
         regs.edi = (uint32_t)dpi;
+        regs.si = flags;
```

Comment:
I doubt this is correct (`si` gets overwritten at `0x4474D7`). Why did you add this?

Reply:
`0x004CF581` and `0x004CF58F` are setting the register, I'm just passing it on.

Reply:
They call the `_inset` version though.

Reply:
I see what you mean now. You're right, they did not need to be added.